### ISSUE TITLE
Questions about the mixup strategy

### ISSUE THREAD

**Issue by yananchen1989** [dsfsi/textaugment#17]
Hello, 

I dive into the mixup code here.
https://github.com/dsfsi/textaugment/blob/master/examples/mixup_example_using_IMDB_sentiment.ipynb 
first of all, mixup does not improve the validation loss and validation accuracy. 
I try by myself, here is the result.
![image](https://user-images.githubusercontent.com/26405281/114492169-9e2a0900-9c4a-11eb-9653-408348ccbccc.png)

Another question is that the mixup seems wrong.
in the code https://github.com/dsfsi/textaugment/blob/master/textaugment/mixup.py

```

lam_vector = np.random.beta(alpha, alpha, batch_size)
index = np.random.permutation(batch_size)
mixed_x = (x.T * lam_vector).T + (x[index, :].T * (1.0 - lam_vector)).T
output_x.append(mixed_x)
if y is None:
    return np.concatenate(output_x, axis=0)
mixed_y = (y.T * lam_vector).T + (y[index].T * (1.0 - lam_vector)).T
output_y.append(mixed_y)
np.concatenate(output_x, axis=0), np.concatenate(output_y, axis=0)
```
the x is the word ids of the original sentence. And it is wrong to mix the word ids via lam_vector, which is generated from beta distribution.

It should be the word embeddings or the sentence embeddings that are about to be mixed up within each batch.

**Comment by vukosim** [comment]
Can you provide the learning curves for with and without mixup.

**Comment by vukosim** [comment]
Hey @yananchen1989, still waiting.

**Comment by yananchen1989** [comment]
Sorry I abandoned the experiments and the records have not been kept. 
I suppose you can check the code from the perspective of logic.
